apis/core/lib/handlers/cube-projects/import.ts

```diff
@@ -27,6 +31,22 @@ export const postImportedProject = protectedResource(
       user,
       userName,
     })
+    res.locals.project = project
+    res.locals.importedDataset = importedDataset
+    next()
+  }),
+  shaclValidate.override({
+    async parseResource(req, res) {
+      const { project, importedDataset } = res.locals
+      return clownface({ dataset: importedDataset }).node(project.id)
+    },
+    async loadShapes() {
+      return $rdf.dataset().import(fromFile(path.resolve(__dirname, '../../domain/cube-projects/import.ttl')))
+    },
```

Comment:
That said, the current implementation of shacl middleware is a little strange in how it assumes targets based on the request.

Thus, a dedicated import shape, loaded explicitly, may be easier to manage

Reply:
I would pre-import all the application-related shapes, and reuse them when adequate.